src/include/network/lan/hardware.rb

```diff
@@ -829,7 +828,7 @@ def storeHW(_key, _event)
         end
         if LanItems.type == "vlan"
           # for vlan devices named vlanN pre-set vlan_id to N, otherwise default to 0
-          LanItems.vlan_id = "#{nm["vlan".size].to_i}"
+          LanItems.vlan_id = nm[VLAN_SIZE].to_i.to_s
```

Comment:
Take into account that this pull request targets storage-ng. You are viewing that change here simply because the pull request includes a merge from master to sync both branches.

The code you are pointing should be fixed in master, not here. I will do it as payment for your LGTM here. :wink: